x/gov/keeper.go

```diff
@@ -280,205 +115,160 @@ func (keeper Keeper) setTallyParams(ctx sdk.Context, tallyParams TallyParams) {
-		err := keeper.ck.SendCoins(ctx, DepositedCoinsAccAddr, BurnedDepositCoinsAccAddr, deposit.Amount)
-		if err != nil {
-			panic("should not happen")
-		}
+	iterator := sdk.KVStorePrefixIterator(store, types.DepositsKeyPrefix)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var deposit types.Deposit
+		keeper.cdc.MustUnmarshalBinaryLengthPrefixed(iterator.Value(), &deposit)
 
-		store.Delete(depositsIterator.Key())
+		if cb(deposit) {
+			break
+		}
 	}
 }
 
-// ProposalQueues
+// IterateDeposits iterates over the all the proposals deposits and performs a callback function
+func (keeper Keeper) IterateDeposits(ctx sdk.Context, proposalID uint64, cb func(deposit types.Deposit) (stop bool)) {
+	iterator := keeper.GetDepositsIterator(ctx, proposalID)
 
-// Returns an iterator for all the proposals in the Active Queue that expire by endTime
-func (keeper Keeper) ActiveProposalQueueIterator(ctx sdk.Context, endTime time.Time) sdk.Iterator {
-	store := ctx.KVStore(keeper.storeKey)
-	return store.Iterator(PrefixActiveProposalQueue, sdk.PrefixEndBytes(PrefixActiveProposalQueueTime(endTime)))
-}
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var deposit types.Deposit
+		keeper.cdc.MustUnmarshalBinaryLengthPrefixed(iterator.Value(), &deposit)
 
-// Inserts a ProposalID into the active proposal queue at endTime
-func (keeper Keeper) InsertActiveProposalQueue(ctx sdk.Context, endTime time.Time, proposalID uint64) {
-	store := ctx.KVStore(keeper.storeKey)
-	bz := keeper.cdc.MustMarshalBinaryLengthPrefixed(proposalID)
-	store.Set(KeyActiveProposalQueueProposal(endTime, proposalID), bz)
+		if cb(deposit) {
+			break
+		}
+	}
 }
 
-// removes a proposalID from the Active Proposal Queue
-func (keeper Keeper) RemoveFromActiveProposalQueue(ctx sdk.Context, endTime time.Time, proposalID uint64) {
+// IterateAllVotes iterates over the all the stored votes and performs a callback function
+func (keeper Keeper) IterateAllVotes(ctx sdk.Context, cb func(vote types.Vote) (stop bool)) {
 	store := ctx.KVStore(keeper.storeKey)
-	store.Delete(KeyActiveProposalQueueProposal(endTime, proposalID))
+	iterator := sdk.KVStorePrefixIterator(store, types.VotesKeyPrefix)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var vote types.Vote
+		keeper.cdc.MustUnmarshalBinaryLengthPrefixed(iterator.Value(), &vote)
+
+		if cb(vote) {
+			break
+		}
+	}
 }
 
-// Returns an iterator for all the proposals in the Inactive Queue that expire by endTime
-func (keeper Keeper) InactiveProposalQueueIterator(ctx sdk.Context, endTime time.Time) sdk.Iterator {
-	store := ctx.KVStore(keeper.storeKey)
-	return store.Iterator(PrefixInactiveProposalQueue, sdk.PrefixEndBytes(PrefixInactiveProposalQueueTime(endTime)))
+// IterateVotes iterates over the all the proposals votes and performs a callback function
+func (keeper Keeper) IterateVotes(ctx sdk.Context, proposalID uint64, cb func(vote types.Vote) (stop bool)) {
```

Comment:
I would change `cb` name with something that simplify the understanding of his purpose and or describe it in the comment

Reply:
`cb` is the callback function that's described on the comment

Reply:
Exactly, I would give to this callback function a name that reminds us what's his pourpose

Reply:
I feel the name `cb` is pretty universal. It's used in Javascript/Node.js too

Reply:
Yeah I think `cb` or `handler` suffices here 👍

Reply:
Why not calling all of them `done` it's another universal one but helps a bit more.
It's not only related to this PR, everywhere we are using `fn`, `handle` or `cb` for this case, and in all cases what the function does is stop the for loop (plus his callbacky stuff)

Reply:
> Why not calling all of them done it's another universal one but helps a bit more.
It's not only related to this PR, everywhere we are using fn, handle or cb for this case, and in all cases what the function does is stop the for loop (plus his callbacky stuff)

I think that question should be addressed as part of #3386